### PR TITLE
Boss_04

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Boss_04/z_boss_04.c [hunk 1]
```diff
@@ -69,75 +74,775 @@ const ActorInit Boss_04_InitVars = {
...
+                func_800B7298(globalCtx, &this->actor, 6);
+                func_80165690();
+                gSaveContext.eventInf[6] |= 1;
+            }
+            break;
+    }
+
+    if (this->unk_70A != 0) {
+        Vec3f sp50;
+        ShrinkWindow_SetLetterboxTarget(0x1B);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Boss_04/z_boss_04.c:401]:
```suggestion
        Vec3f sp50;

        ShrinkWindow_SetLetterboxTarget(0x1B);
```

File: src/overlays/actors/ovl_Boss_04/z_boss_04.c [hunk 2]
```diff
@@ -69,75 +74,775 @@ const ActorInit Boss_04_InitVars = {
...
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_809ED45C(Boss04* this, GlobalContext* globalCtx) {
+    ColliderJntSphElement* temp_v0;
+    u8 damage;
+
+    if ((this->unk_1FE == 0) && (this->collider1.elements->info.bumperFlags & 2)) {
+        this->collider1.elements->info.bumperFlags &= ~2;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Boss_04/z_boss_04.c:569]:
```suggestion
    if ((this->unk_1FE == 0) && (this->collider1.elements->info.bumperFlags & BUMP_HIT)) {
        this->collider1.elements->info.bumperFlags &= ~BUMP_HIT;
```

**Review by AngheloAlf — COMMENTED** [review]
Good work. Just some minor comments

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Boss_04/z_boss_04.c [hunk 3]
```diff
@@ -69,75 +74,776 @@ const ActorInit Boss_04_InitVars = {
...
+                this->unk_2C8 = 300;
+                this->unk_2D0 = 0.0f;
+                D_809EE4D0 = 1;
+                this->unk_2E2 = 60;
+                this->unk_2E0 = 93;
+            }
+
+            if (this->unk_704 > 140) {
+                Camera* sp5C = Play_GetCamera(globalCtx, MAIN_CAM);
+                this->unk_708 = 0;
```
Comment (hensldm) [inline src/overlays/actors/ovl_Boss_04/z_boss_04.c:384]:
```suggestion
                Camera* sp5C = Play_GetCamera(globalCtx, MAIN_CAM);
                
                this->unk_708 = 0;
```

File: src/overlays/actors/ovl_Boss_04/z_boss_04.c [hunk 4]
```diff
@@ -69,75 +74,776 @@ const ActorInit Boss_04_InitVars = {
...
+        sp50.y += Math_SinS(this->unk_748 * 0x4000) * this->unk_748 * 1.5f;
+        Play_CameraSetAtEye(globalCtx, this->unk_70A, &sp50, &this->unk_70C);
+        func_80169940(globalCtx, this->unk_70A, this->unk_744);
+        Math_ApproachF(&this->unk_744, 60.0f, 0.1f, 1.0f);
+    }
+    this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+    x = player->actor.world.pos.x - this->actor.world.pos.x;
+    y = player->actor.world.pos.y - this->actor.world.pos.y;
+    z = player->actor.world.pos.z - this->actor.world.pos.z;
+    this->actor.shape.rot.x = Math_Atan2S(-(y), sqrtf(SQ(x) + SQ(z)));
```
Comment (hensldm) [inline src/overlays/actors/ovl_Boss_04/z_boss_04.c:416]:
```suggestion
    this->actor.shape.rot.x = Math_Atan2S(-y, sqrtf(SQ(x) + SQ(z)));
```

File: src/overlays/actors/ovl_Boss_04/z_boss_04.c [hunk 5]
```diff
@@ -69,75 +74,776 @@ const ActorInit Boss_04_InitVars = {
...
+            this->unk_6C8.x = randPlusMinusPoint5Scaled(600.0f) + this->unk_6E8;
+            this->unk_6C8.z = randPlusMinusPoint5Scaled(600.0f) + this->unk_6F0;
+        }
+    }
+
+    Math_ApproachS(&this->actor.world.rot.y,
+                   Math_Atan2S(this->unk_6C8.x - this->actor.world.pos.x, this->unk_6C8.z - this->actor.world.pos.z), 5,
+                   0x200);
+
+    if (((s8)this->actor.colChkInfo.health < 11) || (KREG(88) != 0)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Boss_04/z_boss_04.c:450]:
```suggestion
    if (((s8)this->actor.colChkInfo.health <= 10) || (KREG(88) != 0)) {
```
Small nit, but round number is probably preferred.

**Review by hensldm — APPROVED** [review]
Just a few cosmetic suggestions.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**